devito/equation.py

```diff
@@ -87,9 +96,26 @@ def evaluate(self):
             except AttributeError:
                 if bool(rules):
                     eq = eq.xreplace(rules)
-
         return eq
 
+    @property
+    def _flatten(self):
+        """
+        Flatten vectorial/tensorial Equation into list of scalar Equations.
+        """
+        try:
+            # Maps the Equations to retreive the rhs from relevant lhs
+            eqs = {lhs: rhs for lhs, rhs in zip(as_tuple(self.lhs), as_tuple(self.rhs))}
+            # get the relevant equations from the lhs structure, .values removes
+            # the symetric duplicates and off-diagonal zeros
+            lhss = self.lhs.values()
+            return [self.func(l, eqs[l], subdomain=self.subdomain,
```

Comment:
Maybe for another time

Reply:
but the list comprehension above looks more pythonic than the (useless?) `eqs` dictionary

Reply:
>c than the (useless?) eqs dictionary

Again it is not useless, what you pasted above may be "prettier" but it does not do what it is supposed to and actually contains useless duplicates. This is what the `.values()` takes care of filtering out the usefull equation and the dictionary keeps the map to the correct associated rhs.

I don't  care if it 's less "pretty" is the pretty way doesn't do what it is supposed to.

Reply:
> I don't care if it 's less "pretty" is the pretty way doesn't do what it is supposed to.

what? really?

Reply:
"if the nice way doesn't work" is the important point

Reply:
It's not about prettiness, it's about heaviness. Clearly before I was kidding.

It might sound stupid and useless, but ensuring that each single line is as simple and as clean as possible eventually pays off -- significantly -- in large codebases , because it speeds up lots of things

iI read those lines a few times and I still couldn't see it, something deceived me. Now I can see it